File: backend/lib/tenantee_web/views/preferences.ex
defmodule TenanteeWeb.PreferencesView do
  def render("show.json", %{preferences: preferences}) do
    %{preferences: Enum.map(preferences, &render("show.json", %{preference: &1}))}
  end

  def render("show.json", %{preference: preference}) do
    %{
      name: preference.name,
      value: preference.value
    }
  end
end

File: frontend/src/utils/preferences.ts
import { Preference } from '../types/preferences';

export function getPreferenceValueByName(
    preferences: Preference[],
    name: string,
    defaultValue: string = '',
): string {
    return (
        preferences.find((preference) => preference.name === name)?.value ??
        defaultValue
    );
}
